packages/vite-plugin-svelte/src/handleHotUpdate.ts

```diff
@@ -62,26 +63,41 @@ export async function handleHotUpdate(
   return result
 }
 
+function normalizeNonCss(code: string, cssHash: string) {
+  // trim HMR transform
+  const indexHmrTransform = code.indexOf(
+    'import * as ___SVELTE_HMR_HOT_API from'
+  )
+  if (indexHmrTransform !== -1) code = code.slice(0, indexHmrTransform)
```

Comment:
this assumes the indexHmrTransform is always at the start, tying this logic to impl in svelte-hmr

Reply:
This assume the transform is at the end, but yeah it ties it horribly to an implementation detail in svelte-hmr. That's why I believe this function should live in svelte-hmr. I'm in the process of doing that and PR an update.